### PR TITLE
feat(List/TPSVC-1580): move 'select all' checkbox to the toolbar

### PR DESCRIPTION
**Pull request by mtikhinskaya**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/List/List.component.js [hunk 1]
```diff
@@ -62,12 +62,24 @@ function List({ id, displayMode, toolbar, list }) {
...
+				items: list.items,
+				isSelected: list.itemProps.isSelected,
+				onToggleAll: list.itemProps.onToggleAll,
+			};
+		}
 	}
 	return (
 		<div className="tc-list">
-			{toolbar && (<Toolbar id={id} {...toolbar} />)}
+			{toolbar && (<Toolbar {...toolbarProps} />)}
```
Comment (sdwvit) [inline src/List/List.component.js]: toolbarProps is undefined here in case if !!toolbar===false

Comment (sdwvit) [inline src/List/List.component.js]: maybe initialize it with {} ? and change 
  toolbarProps = Object.assign({}, toolbar, { id });
to
  toolbarProps = Object.assign(toolbarProps, toolbar, { id });

What do you think?

File: src/List/Toolbar/SelectAll/SelectAll.component.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import theme from './SelectAll.scss';
+
+function SelectAll({ id, items, isSelected, onToggleAll }) {
+	const isAllSelected = () => {
+		const selected = items.reduce((sum, item) => {
```
Comment (sdwvit) [inline src/List/Toolbar/SelectAll/SelectAll.component.js]: please replace with for loop and break as first non-selected item appears
if the list is very long, this reduce is unnecessary long

File: src/List/Toolbar/SelectAll/SelectAll.component.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	const checkboxId = id && `${id}-check-all`;
+	return (
+		<form className="navbar-form navbar-left">
+			<div className={classNames('navbar-text', theme['tc-list-toolbar-select-all'])}>
+				<input
+					id={checkboxId}
+					type="checkbox"
+					onChange={(event) => { onToggleAll(event, items); }}
+					checked={isAllSelected()}
+					disabled={items.length === 0}
```
Comment (sdwvit) [inline src/List/Toolbar/SelectAll/SelectAll.component.js]: fyi !items.length does the same

File: src/List/List.component.js (hunk 1 above)
Comment (mtikhinskaya) [inline src/List/List.component.js]: If `!!toolbar === false` then component will be null. The `&&` in this operation is mean that when first argument is false the result will be null

Comment (sdwvit) [inline src/List/List.component.js]: hm ok, so nothing will be rendered

File: src/List/Toolbar/SelectAll/SelectAll.component.js (hunk 2 above)
Comment (mtikhinskaya) [inline src/List/Toolbar/SelectAll/SelectAll.component.js]: Done

File: src/List/Toolbar/SelectAll/SelectAll.component.js (hunk 3 above)
Comment (mtikhinskaya) [inline src/List/Toolbar/SelectAll/SelectAll.component.js]: Done

File: src/List/Toolbar/SelectAll/SelectAll.component.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import theme from './SelectAll.scss';
+
+function SelectAll({ id, items, isSelected, onToggleAll }) {
+	const isAllSelected = () => items.findIndex(item => !isSelected(item)) < 0;
```
Comment (sdwvit) [inline src/List/Toolbar/SelectAll/SelectAll.component.js]: 👌

**Pull request merged**